core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -1319,4 +1328,26 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
     tsm.handleFailedTask(tsm.taskAttempts.head.head.taskId, TaskState.FAILED, TaskKilled("test"))
     assert(tsm.isZombie)
   }
+  test("SPARK-25250 On successful completion of a task attempt on a partition id, kill other" +
```

Comment:
Nit: blank line above, and please use spaces in the body of the test to break this up into more readable chunks.

Reply:
Done